labs/b6.md

```diff
@@ -0,0 +1,153 @@
+
+If you are interested in learning about the different signals, run `man 7 signal`.
+
+## Cron
+So much infrastructure in the computing world relies on scheduled processes. This is the job of the cron daemon. For example, at the OCF, we use a cron job that runs every thirty minutes to keep our servers and desktops updated via Puppet. To get a feel for the cron scheduler, we're going to write a basic cron job.
+
+Open the cron editor by running `crontab -e` (if the editor of your choice isn't being launched, set the `EDITOR` environment variable), which will create a crontab for your user. Here is a sample task:
+
+```* * * * * date +"\%T" >> $HOME/timestamps.txt```
+
+Right now this runs every minute. Modify it to run every five minutes. Make sure to leave this running while you complete the remainder of the lab as you will be turning this file in.
+
+If you get stuck, visit [crontab.guru](https://crontab.guru)!
+
+## Using systemd
+
+### What services are running right now?
+Run `systemctl`. You’ll see a long table of every unit known to systemd.
+
+Let’s narrow it down to services for now. Run `systemctl --type=service`. Now you can see a list of all services running on your computer. Each of these services is a daemon running in the background. Do you see any familiar services running?
+
+### Controlling Services
+Now let's use `systemd` to control a an nginx web server. Install nginx by issuing `sudo apt install nginx`. Once that is done we can tell systemd to start the service with the following: `sudo systemctl start nginx`. Run `systemctl status nginx` to ensure it is running and navigate to http://*yourvm*.decal.xcf.sh/ -- you should be greeted by the nginx default landing page.
+
+Now let's make nginx listen for connections on the nonstandard port 420. In `/etc/nginx/sites-available/default` change the following lines:
+
+```nginx
+listen 80 default_server;
+listen [::]:80 default_server;
+```
+
+to:
+```nginx
+listen 420 default_server;
+listen [::]:420 default_server;
+```
+
+Tell systemd that nginx has changed configuration and needs reloading with: `sudo systemctl reload nginx`. Now, accessing  http://*yourvm*.decal.xcf.sh/ should now give you a connection refused error and your webserver will only be accessible via http://*yourvm*.decal.xcf.sh:420/.
+
+Note that not all services can be reloaded; systemd will notify you if this is the case and such services will have to be restarted instead: `sudo systemctl restart yourservice`.
+
+Finally go ahead and stop the nginx service. What is the command to do this?
+
+### Creating a service
+Let's set up a web server and create a systemd unit for it.
+
+To get the code run:
+```wget -qO- https://www.ocf.berkeley.edu/~keur/decal/install.sh | bash && cd toy```
```

Comment:
puppet this or put it in `decal-labs`, also, maybe we shouldn't teach students that curling things into bash is a good pattern? idk

Reply:
It's okay to pipe into bash if it's your teacher :)